### PR TITLE
fix TR5 hydra power up flame and missile shockwave

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**
just a small fix for the TR5 hydra
fixed shockwave light for hammer romangod

### CODE REVIEW

**Comment by Stranger1992** [comment]
I tested both the `HYDRA` and `ROMAN_GOD1` .

I didn't spot anything with the Roman God and I really welcome the additional dynamic light on the ground attack.

The hydra's charge attack looks great (directly compared with the the original TRC). My only problem is that it took a VERY long time to kill the Hydra. I counted it took approximately 60 revolver shots to kill it, which is more powerful than any other enemy. I think that can be reduced but other than that it is good to go!

File: TombEngine/Game/effects/effects.cpp [hunk 1]
```diff
@@ -1231,6 +1231,15 @@ void KillAllCurrentItems(short itemNumber)
 	// TODO: Reimplement this functionality.
 }
 
+void SpawnDynamicLight(const Vector3& pos, const Color& color, float falloff)
```
Comment (Lwmte) [inline TombEngine/Game/effects/effects.cpp:1234]: Why this function is called `SpawnDynamicLight`, if we have a function called `TriggerDynamicLight` already? Make an overload.

File: TombEngine/Objects/TR4/Entity/tr4_demigod.cpp [hunk 2]
```diff
@@ -27,6 +27,14 @@ namespace TEN::Entities::TR4
 	constexpr auto DEMIGOD1_WALK_RANGE					   = SQUARE(BLOCK(3));
 	constexpr auto DEMIGOD2_RADIAL_PROJECTILE_ATTACK_RANGE = SQUARE(BLOCK(5));
 	constexpr auto DEMIGOD3_RADIAL_PROJECTILE_ATTACK_RANGE = SQUARE(BLOCK(5));
+	constexpr auto DEMIGOD1_SHOCKWAVE_LIGHT_DURATION = 1.0f;
+
+	struct Demigod1Info
```
Comment (Lwmte) [inline TombEngine/Objects/TR4/Entity/tr4_demigod.cpp:32]: What is this?

File: TombEngine/Objects/TR4/Entity/tr4_demigod.cpp [hunk 3]
```diff
@@ -27,6 +27,14 @@ namespace TEN::Entities::TR4
...
+	{
+		Vector3i LightPosition = Vector3i::Zero;
+	};
+
+	Demigod1Info Demigod1Data;
```
Comment (Lwmte) [inline TombEngine/Objects/TR4/Entity/tr4_demigod.cpp:37]: Why such complexity? You make a struct which keeps a vector3. You define a global variable which is this struct. Then you refer to this struct only to pass a vector3 from it to `SpawnDynamicLight`. Can't you just directly use this position to spawn it?

File: TombEngine/Objects/TR4/Entity/tr4_demigod.cpp [hunk 4]
```diff
@@ -709,6 +730,15 @@ namespace TEN::Entities::TR4
 					TriggerShockwave((Pose*)&pos, 24, 88, 256, 128, 128, 128, 32, EulerAngles::Zero, 8, true, false, (int)ShockwaveStyle::Normal);
 					TriggerHammerSmoke(pos.x, pos.y + 128, pos.z, 8);
 
+					Demigod1Data.LightPosition = Vector3i(pos.x, pos.y, pos.z);
```
Comment (Lwmte) [inline TombEngine/Objects/TR4/Entity/tr4_demigod.cpp:733]: As I said above, you should not use a global.

**Review by Lwmte — CHANGES_REQUESTED** [review]

File: TombEngine/Game/effects/effects.cpp (hunk 1 above)
Comment (Sezzary) [reply TombEngine/Game/effects/effects.cpp:1234]: `SpawnDynamicLight()` is a replacement for `TriggerDynamicLight()` that uses the newer, consistently applied prefix convention decided upon for new or revised effect functions (`Spawn`).

Comment (Lwmte) [reply TombEngine/Game/effects/effects.cpp:1234]: I think you can do en masse rename in other PR then. Having 2 functions with same purpose but named differently is misleading.

File: TombEngine/Objects/TR4/Entity/tr4_demigod.cpp (hunk 4 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/TR4/Entity/tr4_demigod.cpp]: fixed

File: TombEngine/Objects/TR4/Entity/tr4_demigod.cpp (hunk 3 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/TR4/Entity/tr4_demigod.cpp]: fixed

File: TombEngine/Objects/TR4/Entity/tr4_demigod.cpp (hunk 2 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/TR4/Entity/tr4_demigod.cpp]: fixed

**Pull request merged**